File: src/matlab/com/springbok/antenna/AntennaTest.m
classdef AntennaTest < TestUtility
% Tests methods of Antenna class.
  
% Copyright (C) 2016 Springbok LLC <http://springbok.io/> All rights reserved.

  properties (Constant = true)
    
    % Antenna name
    name = 'RAMBOUILLET';
    % Antenna gain [dB]
    gain = 54.000000000000000;

    % Antenna feeder loss [db]
    feeder_loss = 0;
    % Antenna body loss [db]
    body_loss = 0;
    % Antenna noise temperature [K]
    noise_t = NaN;
    % Antenna unit azimuth reference (normal) vector in local tangent coordinates
    x_ltp = [];
    % Antenna unit elevation reference vector in local tangent coordinates
    z_ltp = [];
    
  end % properties (Constant = true)
  
  properties (Access = private)
    
    % An antenna
    antenna
    
  end % properties (Access = private)
  
  methods
    
    function this = AntennaTest(logFId, testMode)
    % Constructs an AntennaTest.
    %
    % Parameters
    %   logFId - Log file identifier
    %   testMode - Test mode, if 'interactive' then beeps and pauses
      
    % Invoke the superclass constructor
      if nargin == 0
        superArgs = {};
        
      else
        superArgs{1} = logFId;
        superArgs{2} = testMode;
        
      end % if
      this@TestUtility(superArgs{:});
      
      % Compute derived properties
      this.antenna = Antenna(this.name, ...
                             this.gain);

    end % AntennaTest()
    
    function test_Antenna(this)
    % Tests Antenna method.
      
      t = [];
      t = [t; strcmp(this.antenna.name, this.name)];
      t = [t; isequal(this.antenna.gain, this.gain)];
      t = [t; isequal(this.antenna.feeder_loss, this.feeder_loss)];
      t = [t; isequal(this.antenna.body_loss, this.body_loss)];
      t = [t; isequalwithequalnans(this.antenna.noise_t, this.noise_t)];
      t = [t; isequalwithequalnans(this.antenna.x_ltp, this.x_ltp)];
      t = [t; isequalwithequalnans(this.antenna.z_ltp, this.z_ltp)];

      this.assert_true( ...
          'Antenna', ...
          'Antenna', ...
          this.IS_EQUAL_DESC, ...
          min(t));
      
    end % test_Antenna()
    
  end % methods
  
end % classdef

File: src/java/com/springbok/antenna/AntennaTest.java
/* Copyright (C) 2022 Springbok LLC

This program is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or (at
your option) any later version.

This program is distributed in the hope that it will be useful, but
WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program.  If not, see <https://www.gnu.org/licenses/>.
*/
package com.springbok.antenna;

import static org.junit.Assert.*;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;

/**
 * Tests methods of Antenna class.
 * 
 * @author raymondleclair
 */
public class AntennaTest {

	// Antenna name
	private final String name = "RAMBOUILLET";
	// Antenna gain [dB]
	private final double gain = 54.0;
	// Antenna noise temperature [K]
	private final double noise_t = Double.NaN;

	// An antenna
	private Antenna antenna;

	@Before
	public void setUp() throws Exception {

		// Compute derived properties
		this.antenna = new Antenna(this.name, this.gain, this.noise_t);
	}

	@After
	public void tearDown() throws Exception {
	}

	@Test
	/*
	 * Tests Antenna method.
	 */
	public void test_Antenna() {
		assertEquals(this.antenna.get_name(), this.name);
		assertEquals(this.antenna.get_gain(), this.gain, 1.0e-16);
		assertTrue(Double.isNaN(this.antenna.get_noise_t()));
	}
}
